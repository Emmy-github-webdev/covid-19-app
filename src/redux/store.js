import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';

const reducer = combineReducers({

});

const initialState = {};

const middleware = [thunk, logger];

const store = createStore(
  reducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

export default store;
