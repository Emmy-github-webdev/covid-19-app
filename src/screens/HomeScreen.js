import React from 'react';

const HomeScreen = () => (
  <div>
    Home screen
  </div>
);

export default HomeScreen;
