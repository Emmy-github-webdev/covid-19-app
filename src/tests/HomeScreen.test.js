import { Provider } from "react-redux";
import '@testing-library/jest-dom';
import {render, screen, cleanup, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import HomeScreen from "../screens/HomeScreen";

afterEach(() => {
  cleanup();
});

describe('Testing redux store connection', () => {
  
  it('Render home screen without error', () => {
    render(<Provider store={store}><HomeScreen /></Provider>);
    const homeElement = screen.getByTestId('home');
    expect(homeElement).toBeInTheDocument();
  });

  test('Testing store', () => {
    const home = renderer.create(<Provider store={store}><HomeScreen /></Provider>).toJSON();
    expect(home).toMatchSnapshot();
  });

});