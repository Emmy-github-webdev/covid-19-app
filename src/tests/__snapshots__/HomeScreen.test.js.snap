// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing redux store connection Testing store 1`] = `
<div
  data-testid="home"
>
  <div>
    <nav
      className="nav"
      data-testid="navbar"
    >
      <div
        className="year-wrapper"
      >
        <svg
          color="white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
        2022
      </div>
      <div
        className="title-wrapper"
      >
        <h1>
          Africa
        </h1>
      </div>
      <div
        className="icons-wrapper"
      >
        <div
          className="icon-one"
        >
          <svg
            color="white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
            />
          </svg>
        </div>
        <div
          className="icon-two"
        >
          <svg
            color="white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="hero-wrapper container-fluid"
  >
    <div
      className="hero-img-display"
    >
      <img
        alt="map of africa"
        src="africa-map.svg"
      />
    </div>
    <div
      className="header-wrapper"
    >
      <h1>
        Africa
      </h1>
      <h3>
         0 Vaccinated
      </h3>
    </div>
  </div>
  <div
    className="sorted-bar"
  >
    <h1>
      STATS BY COUNTRY
    </h1>
  </div>
  <div
    className="home-wrapper row g-0"
  />
</div>
`;
